### PR TITLE
Fix flaky cert tests

### LINKED ISSUE #3266: flakey unit test: Test_server_ClientCert/*
## Flaky Test

Effects main, example [here](https://buildkite.com/elastic/fleet-server/builds/3172#018d8e62-141f-4688-a2b6-5e67b179e980)
File [here](https://github.com/elastic/fleet-server/blob/main/internal/pkg/api/server_test.go)
It looks like it is where the request may get a connection refused response

### Stack Trace

```
=== RUN   Test_server_ClientCert/valid_client_certs_no_certs_requested
--
  | server_test.go:349:
  | Error Trace:	/opt/buildkite-agent/builds/bk-agent-prod-gcp-1707490899215587797/elastic/fleet-server/internal/pkg/api/server_test.go:349
  | Error:      	Received unexpected error:
  | Get "https://localhost:36805/api/status": dial tcp 127.0.0.1:36805: connect: connection refused
  | Test:       	Test_server_ClientCert/valid_client_certs_no_certs_requested
```

### PR DESCRIPTION
**Pull request by michel-laterman**
## What is the problem this PR solves?

Fix flaky mTLS tests by ensuring the server runs for 500ms without any errors before any interactions are made with it.
We also filter out context cancelation errors returned from server.Run so they are not checked and only validate no errors were returned after the wait group is done

## Related issues

- Closes #3266

### CODE REVIEW

**Comment by michel-laterman** [comment]
I'm going to manually retrigger the build a few times to see if it fails

**Comment by michel-laterman** [comment]
buildkite test this

**Comment by michel-laterman** [comment]
buildkite test this

**Comment by michel-laterman** [comment]
buildkite test this

**Comment by michel-laterman** [comment]
buildkite test this

File: internal/pkg/api/server_test.go [hunk 1]
```diff
@@ -46,40 +43,38 @@ func Test_server_Run(t *testing.T) {
...
-	pim := mock.NewMockMonitor()
-	pm := policy.NewMonitor(bulker, pim, config.ServerLimits{PolicyLimit: config.Limit{Interval: 5 * time.Millisecond, Burst: 1}})
-	bc := checkin.NewBulk(nil)
-	ct := NewCheckinT(verCon, cfg, c, bc, pm, nil, nil, nil, nil)
-	et, err := NewEnrollerT(verCon, cfg, nil, c)
-	require.NoError(t, err)
-
-	srv := NewServer(addr, cfg, ct, et, nil, nil, nil, nil, fbuild.Info{}, nil, nil, nil, nil, nil)
-	errCh := make(chan error)
+	srv := NewServer(addr, cfg, nil, nil, nil, nil, nil, nil, fbuild.Info{}, nil, nil, nil, nil, nil)
```
Comment (nchaulet) [inline internal/pkg/api/server_test.go:46]: @michel-laterman may be you can add a little more details on what was causing the flakyness, why do  we need create the server without bulker/policy monitor, ..?

Comment (michel-laterman) [reply internal/pkg/api/server_test.go:46]: bulker, policy monitor, status checker are not needed in this test case because we don't test any endpoints

**Review by nchaulet — APPROVED** [review]
LGTM

**Pull request merged**